### PR TITLE
Fix spec v2

### PR DESCRIPTION
**Pull request by christinach**
This is checked out from @tpendragon's branch https://github.com/pulibrary/orangelight/commit/92fb95e73ac25e66a76c3c07522d8211fe2d8b5b

also part of #2649

### CODE REVIEW

**Comment by christinach** [comment]
Discussed this in the checkin 08-16-2021. Will generate json from traject and update the fixtures.

**Comment by christinach** [comment]
@hectorcorrea pointed out https://github.com/traject/traject#readers-and-writers

File: spec/features/holding_locations_sort_order_spec.rb [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # frozen_string_literal: true
 
 require 'rails_helper'
-
+# See issue https://github.com/pulibrary/orangelight/issues/2660
+# Do we still need this sort when there are recap holdings?
```
Comment (christinach) [inline spec/features/holding_locations_sort_order_spec.rb:5]: opened https://github.com/pulibrary/orangelight/issues/2660

File: spec/features/request_options_spec.rb [hunk 2]
```diff
@@ -119,55 +108,4 @@
       end
     end
   end
-
-  describe 'An In-transit discharged item', js: true do
```
Comment (christinach) [inline spec/features/request_options_spec.rb:123]: in transit is not a status that we have anymore

File: spec/features/request_options_spec.rb [hunk 3]
```diff
@@ -119,55 +108,4 @@
...
-    it 'does not display a request button', unless: in_ci? do
-      using_wait_time 5 do
-        expect(page.all('.holding-block').length).to eq 1
-        expect(page).to have_selector('td[data-requestable="false"]')
-      end
-    end
-  end
-
-  describe 'Paging location available status', js: true do
-    before do
```
Comment (christinach) [inline spec/features/request_options_spec.rb:137]: Paging location could be replaced with 'Pick up' but we are checking in other tests for the 'Request' button.

File: spec/models/solr_document_spec.rb [hunk 4]
```diff
@@ -416,17 +416,18 @@
...
+    # See https://github.com/pulibrary/orangelight/issues/2661
+    xit 'returns the holdings from the host record if the record is contained' do
+      solr_document = described_class.new(id: "9947055653506421", "contained_in_s": ["99124977073506421"])
       expect(solr_document.holdings_all_display).to eq JSON.parse(host_doc_holdings)
     end
 
-    it 'returns the combined holdings if the record is contained and has its own holdings' do
-      solr_document = described_class.new(id: "9929455793506421", "contained_in_s": ["99121886293506421"], holdings_1display: contained_doc_holdings)
+    xit 'returns the combined holdings if the record is contained and has its own holdings' do
+      solr_document = described_class.new(id: "9929455793506421", "contained_in_s": ["99124994093506421"], holdings_1display: contained_doc_holdings)
```
Comment (christinach) [inline spec/models/solr_document_spec.rb:430]: Opened https://github.com/pulibrary/orangelight/issues/2661

File: spec/features/request_options_spec.rb [hunk 5]
```diff
@@ -3,71 +3,46 @@
...
-      using_wait_time 5 do
-        click_link('Request')
-        expect(current_path).to eq "/requests/9618072"
-      end
+      visit '/catalog/99113436223506421'
     end
-  end
-
-  describe 'Available status non-requestable location', js: true do
-    before do
```
Comment (christinach) [inline spec/features/request_options_spec.rb:30]: All the locations in bibdata are requestable: true

File: spec/views/catalog/show.html.erb_spec.rb [hunk 6]
```diff
@@ -66,14 +66,14 @@
...
+        expect(page).not_to have_selector('#doc_9960108133506421 > dl > dt.blacklight-holdings_1display')
       end
     end
-
-    it 'is rendered' do
-      visit 'catalog/6010813'
-      expect(page).to have_selector('#doc_6010813 > dl > dt.blacklight-holdings_1display')
+    # This seems to have changed. Maybe because we dont display items anymore in the show page?
+    xit 'is rendered' do
+      visit 'catalog/998574693506421'
```
Comment (tpendragon) [inline spec/views/catalog/show.html.erb_spec.rb:75]: Question: Why is this not `9960108133506421`?

File: spec/controllers/bookmarks_controller_spec.rb [hunk 7]
```diff
@@ -19,14 +19,14 @@
...
-    let(:data2) { ['3148879', 'Sin, Chae-hong', '신재홍', "Tongnipkun u\u0306i cho\u0306nt'u", '독립군의전투　/'] }
+    let(:data1) { ['9997412163506421', 'Adriaen van de Velde : Dutch master of landscape', 'Cornelis, Bart', 'Book', 'English', 'London: Paul Holberton Publishing', '2016', '224 pages : illustrations (some color) ; 29 cm', 'Remote Storage: Marquand Use Only', 'ND653.V414 A4 2016', 'Published to accompany the exhibition held at the Rijksmuseum, Amsterdam, 24 June - 15 September 2016 and Dulwich Picture Gallery, London, 12 October 2016 - 15 January 2017.'] }
+    let(:data2) { ['9935444363506421', 'Replacement migration : is it a solution to declining and ageing populations?', 'Book', 'English', 'New York: United Nations', '2001', 'viii, 151 p. : ill. ; 28 cm.', 'United Nations Collection; Wallace Hall', '01.XIII.19; JV6225 .R464 2001'] }
```
Comment (tpendragon) [inline spec/controllers/bookmarks_controller_spec.rb:23]: Question: Why is this not `9931488793506421`?

File: spec/controllers/bookmarks_controller_spec.rb [hunk 8]
```diff
@@ -19,14 +19,14 @@
...
     let(:bad_value) { 'Adriaen van de Velde : Dutch master of landscape / Bart Cornelis, Marijn Schapelhouman' }
 
     it 'renders a CSV list of metadata' do
       sign_in user
-      user.bookmarks.create!([{ document_id: '9741216', document_type: 'SolrDocument' }])
-      user.bookmarks.create!([{ document_id: '3148879', document_type: 'SolrDocument' }])
+      user.bookmarks.create!([{ document_id: '9997412163506421', document_type: 'SolrDocument' }])
+      user.bookmarks.create!([{ document_id: '9935444363506421', document_type: 'SolrDocument' }])
```
Comment (tpendragon) [inline spec/controllers/bookmarks_controller_spec.rb:29]: Question: Why is this not 9931488793506421?

File: spec/features/browsing_item_spec.rb [hunk 9]
```diff
@@ -31,7 +31,7 @@
 
   context 'when an entry has a bib. ID and ARK for a resource published in Figgy' do
     before do
-      visit 'catalog/3395923'
+      visit 'catalog/9970446223506421'
```
Comment (tpendragon) [inline spec/features/browsing_item_spec.rb:34]: Question: Why is this not 9933959233506421?

File: spec/features/browsing_item_spec.rb [hunk 10]
```diff
@@ -40,7 +40,7 @@
     end
 
     it 'updates the link to the ARK with a fragment identifier for the UV' do
-      expect(page).to have_selector '.electronic-access a[href$="3395923#view"]', text: 'Digital content'
+      expect(page).to have_selector '.availability--online a[href$="7044622#view"]', text: 'Table of contents'
```
Comment (tpendragon) [inline spec/features/browsing_item_spec.rb:43]: Question: Why did this change?

File: spec/features/browsing_item_spec.rb [hunk 11]
```diff
@@ -8,19 +8,19 @@
...
-    visit 'catalog/3478898'
+    visit 'catalog/9934788983506421'
     expect(page).to have_selector '.icon-share[aria-hidden="true"]', visible: false
     expect(page).to have_selector '.icon-print[aria-hidden="true"]', visible: false
   end
 
   context 'when an entry has a bib. ID for a resource published in Figgy' do
     before do
-      visit 'catalog/3753928'
+      visit 'catalog/9970446223506421'
```
Comment (tpendragon) [inline spec/features/browsing_item_spec.rb:23]: Question: Why not `9937539283506421`?

File: spec/features/citation_spec.rb [hunk 12]
```diff
@@ -3,14 +3,20 @@
 require 'rails_helper'
 
 describe 'citation' do
-  let(:bibid) { '7994866' }
-
+  before do
```
Comment (tpendragon) [inline spec/features/citation_spec.rb:6]: Question: Why extract this to a before? Are there other tests depending on this?

File: spec/features/header_links_spec.rb [hunk 13]
```diff
@@ -7,7 +7,7 @@
 
   context 'With MARC-based records' do
     before do
-      visit '/catalog/10585552'
+      visit '/catalog/99115031773506421'
```
Comment (tpendragon) [inline spec/features/header_links_spec.rb:10]: Question: Why is this not 9910585552?

File: spec/features/header_links_spec.rb [hunk 14]
```diff
@@ -36,7 +36,7 @@
 
   context 'With non-MARC-based records' do
     before do
-      visit  '/catalog/dsp01ft848s955'
+      visit  '/catalog/dsp017s75dc44p'
```
Comment (tpendragon) [inline spec/features/header_links_spec.rb:39]: Question: Why is this changed?

File: spec/features/request_options_spec.rb [hunk 15]
```diff
@@ -3,71 +3,46 @@
...
+      stub_request(:get, "#{Requests.config['bibdata_base']}/availability?id=99113436223506421")
         .to_return(status: 200, body: {}.to_json, headers: {})
-      stub_request(:get, "https://bibdata-staging.princeton.edu/locations/delivery_locations.json")
+      stub_request(:get, "#{Requests.config['bibdata_base']}/locations/delivery_locations.json")
         .to_return(status: 200, body: {}.to_json, headers: {})
-      using_wait_time 5 do
-        click_link('Request')
-        expect(current_path).to eq "/requests/9618072"
-      end
+      visit '/catalog/99113436223506421'
```
Comment (tpendragon) [inline spec/features/request_options_spec.rb:18]: Question: Why is this not 9996180723506421?

File: spec/features/request_options_spec.rb [hunk 16]
```diff
@@ -109,7 +84,7 @@
 
   describe 'Aeon location', js: true do
     before do
-      visit '/catalog/7916044'
+      visit '/catalog/99118600973506421'
```
Comment (tpendragon) [inline spec/features/request_options_spec.rb:87]: Question: Why is this not 9979160443506421?

File: spec/views/catalog/show.html.erb_spec.rb [hunk 17]
```diff
@@ -9,34 +9,34 @@
...
-      expect(page).to have_selector('div#view')
+      visit 'catalog/9961093233506421'
+      expect(page).to have_selector('#view')
     end
   end
 
   context 'when entries describe resources published using multiple ARKs', js: true do
     it 'renders multiple viewers' do
-      visit '/catalog/3943643'
+      visit '/catalog/9970446223506421'
```
Comment (tpendragon) [inline spec/views/catalog/show.html.erb_spec.rb:26]: Question: Why is this not 993943643350421?

File: spec/views/catalog/show.html.erb_spec.rb [hunk 18]
```diff
@@ -66,14 +66,14 @@
   describe 'the location for physical holdings', js: true do
     context 'if physical holding information is recorded in the entry' do
       it 'is not rendered' do
-        visit 'catalog/857469'
-        expect(page).not_to have_selector('#doc_857469 > dl > dt.blacklight-holdings_1display')
+        visit 'catalog/9960108133506421'
```
Comment (tpendragon) [inline spec/views/catalog/show.html.erb_spec.rb:69]: Question: Why is this not 998574693506421?

**Review by tpendragon — COMMENTED** [review]
Most of my questions here are about why the tested records have changed. I have a general concern that the old record and the new record may not be 1-1 matches for the functionality being tested. How sure are you?

File: spec/views/catalog/show.html.erb_spec.rb (hunk 6 above)
Comment (christinach) [reply spec/views/catalog/show.html.erb_spec.rb:75]: This is testing the location for physical holdings. id:https://catalog.princeton.edu/catalog/9960108133506421 has no holdings. 
I tested another id: 998574693506421   to make sure I can see the blacklight class. Which I couldn't so I marked the test pending and added a note. I will open an issue about that.

File: spec/controllers/bookmarks_controller_spec.rb (hunk 7 above)
Comment (christinach) [reply spec/controllers/bookmarks_controller_spec.rb:23]: This is testing the csv table and makes sure the info is there. Which it is. It was easier for me to test English and not Korean.

File: spec/controllers/bookmarks_controller_spec.rb (hunk 8 above)
Comment (christinach) [reply spec/controllers/bookmarks_controller_spec.rb:29]: This is the same test. The csv is generated and the info is there. I preferred to test English and not Korean.

File: spec/features/browsing_item_spec.rb (hunk 9 above)
Comment (christinach) [reply spec/features/browsing_item_spec.rb:34]: id 9933959233506421 was not in the fixtures I had in current_json under alma. The id I'm using is from EAL, a more complicated example with 2 viewers that tests UV ark and thumbnail.

File: spec/features/browsing_item_spec.rb (hunk 10 above)
Comment (christinach) [reply spec/features/browsing_item_spec.rb:43]: I had the same question. I couldn't get to the href from the .electronic-access class. Only from .availability--online class. 

<div class="availability--online">
    <h3>Available Online</h3>
    <ul><ul><li class="electronic-access"></li><li><a href="/catalog/7044622#view">Table of contents</a></li><li class="electronic-access"></li><li><a href="/catalog/7044622#view">Digital content</a></li><li id="full_text" class="availability--panel availability_full-text availability_full-text-alternative" data-umlaut-services="true"></li></ul><li class="electronic-access"></li><li><a target="_blank" href="https://na05.alma.exlibrisgroup.com/view/uresolver/01PRI_INST/openurl?u.ignore_date_coverage=true&amp;portfolio_pid=53837284010006421&amp;Force_direct=true"></a> </li><li class="electronic-access"></li><li><a target="_blank" href="https://na05.alma.exlibrisgroup.com/view/uresolver/01PRI_INST/openurl?u.ignore_date_coverage=true&amp;portfolio_pid=53837284030006421&amp;Force_direct=true"></a> </li></ul>
  </div>

File: spec/features/browsing_item_spec.rb (hunk 11 above)
Comment (christinach) [reply spec/features/browsing_item_spec.rb:23]: It tests for thumbnail and UV. I used the example with the 2 viewers. https://catalog.princeton.edu/catalog/9970446223506421https://figgy.princeton.edu/catalog?utf8=%E2%9C%93&q=9970446223506421 it is in figgy.

File: spec/features/citation_spec.rb (hunk 12 above)
Comment (christinach) [reply spec/features/citation_spec.rb:6]: No there are not for now. I can move it in the it.

File: spec/features/header_links_spec.rb (hunk 13 above)
Comment (christinach) [reply spec/features/header_links_spec.rb:10]: id: 99105855523506421 was not in my fixtures. The id I used is testing for the same features navbar, account dropdown, and record view, widgets, etc.

File: spec/features/request_options_spec.rb (hunk 15 above)
Comment (christinach) [reply spec/features/request_options_spec.rb:18]: The test is testing that Request button that exists. It doesnt make a difference. I changed to the other id when I couldn't make capybara run successfully thinking that maybe it had to do with a delay on the record.

File: spec/features/header_links_spec.rb (hunk 14 above)
Comment (christinach) [reply spec/features/header_links_spec.rb:39]: The dspace fixture I used was in my fixtures. The test is specific to dspace. The record page should not have cite, RefWorks, or EndNote links. https://catalog.princeton.edu/catalog/dsp01ft848s955 
https://catalog.princeton.edu/catalog/dsp017s75dc44p

File: spec/features/request_options_spec.rb (hunk 16 above)
Comment (christinach) [reply spec/features/request_options_spec.rb:87]: The test -- which in a way is redundant because we are testing on top as well for Reading Room Request button -- is looking for Reading room request button which means AEON. Both ids have a Reading Room Request button and will direct the user to Aeon. The id I used was in the updated fixtures.

File: spec/views/catalog/show.html.erb_spec.rb (hunk 17 above)
Comment (christinach) [reply spec/views/catalog/show.html.erb_spec.rb:26]: id 993943643350421 Does not exist in Alma. I used the record with the 2 viewers.

File: spec/views/catalog/show.html.erb_spec.rb (hunk 18 above)
Comment (christinach) [reply spec/views/catalog/show.html.erb_spec.rb:69]: None of the ids have the class. I was switching between the ids to make sure that it is not rendered. I will open an issue about this blacklight class. I think it might be because now we display only the holding information and not the item but I dont know yet.

**Comment by christinach** [comment]
@tpendragon can you check? I moved the stub request in the it. Will also open the issues I mentioned in the comments.

File: spec/views/catalog/show.html.erb_spec.rb (hunk 18 above)
Comment (christinach) [reply spec/views/catalog/show.html.erb_spec.rb:69]: https://github.com/pulibrary/orangelight/issues/2663

File: spec/views/catalog/show.html.erb_spec.rb (hunk 6 above)
Comment (christinach) [reply spec/views/catalog/show.html.erb_spec.rb:75]: https://github.com/pulibrary/orangelight/issues/2663

**Review by tpendragon — APPROVED** [review]
